### PR TITLE
Add samples for Data Catalog lookupEntry

### PR DESCRIPTION
**Pull request by ricardosm-cit**
Adding samples to Data Catalog lookup_entry, according to https://github.com/GoogleCloudPlatform/java-docs-samples/issues/1410

### CODE REVIEW

**Comment by lesv** [comment]
PTAL the new sample format in CONTRIBUTING.md @kurtisvg for additional info.

**Comment by ricardosm-cit** [comment]
> PTAL the new sample format in CONTRIBUTING.md @kurtisvg for additional info.

Thanks @lesv! I read it before submitting the change. Did I miss anything?

**Comment by kurtisvg** [comment]
It's highly suggested to use the updated sample format, which has guidelines [here](https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md). This is to avoid having to update the samples again in the new future.

**Comment by lesv** [comment]
My quick glance was that I didn't see the comments on the parameters being passed, so I didn't scan the rest of the sample.  https://github.com/GoogleCloudPlatform/java-docs-samples/blob/master/SAMPLE_FORMAT.md

And take a look here:  https://github.com/GoogleCloudPlatform/java-docs-samples/tree/master/dlp/src/main/java/dlp/snippets

**Comment by ricardosm-cit** [comment]
Just pushed a new version with several changes. I believe it's more consistent with the recommended standards now.

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+   * Lookup the Data Catalog entry referring to a BigQuery Dataset
+   *
+   * @param projectId The project ID to which the Dataset belongs, e.g. 'my-project'
+   * @param datasetId The dataset ID to which the Catalog Entry refers, e.g. 'my_dataset'
+   * @param lookupBySqlResource Indicates whether the lookup should be performed by Sql Resource
+   *     instead of Linked Resource, e.g. 'false'
+   */
+  private static void lookupEntryBigQueryDataset(
+      String projectId, String datasetId, boolean lookupBySqlResource) {
+
```
Comment (kurtisvg) [inline datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java]: Function parameters should have a commented out declaration that allows the user to copy/paste the variables somewhere else:
```java
// String projectId = "my-project"
// String datasetId = "my_dataset"
```
lookupBySqlResource probably shouldn't be a function parameter - other samples with similar branching parameters (like file extension or something similar) the consensus was that should only show one or be separate files. Not sure of the context here, so I'm not sure what's most appropriate.

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    LookupEntryRequest request;
+
+    // Construct the Lookup request to be sent by the client.
```
Comment (kurtisvg) [inline datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java]:
```suggestion
    // Construct the Lookup request to be sent by the client.
    LookupEntryRequest request;
    if (lookupBySqlResource) {
```

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    if (lookupBySqlResource) {
```
Comment (kurtisvg) [inline datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java]: Please add some additional context around these statements. What is the difference between these two things? Which one should I use in which scenarios? Are these easy to find id's associated with the resource, or is there information available on how I should use them?

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+      System.out.print("Error during lookupEntryBigQueryDataset:\n" + e.toString());
+    }
+  }
+
+  /**
+   * Command line application to lookup the Data Catalog entry referring to a BigQuery Dataset.
+   * Requires 2 positional args: projectId and datasetId. A third arg is optional:
+   * lookupBySqlResource (when set, the lookup is done by Sql Resource instead of Linked Resource).
+   */
+  public static void main(String... args) {
```
Comment (kurtisvg) [inline datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java]: Please remove the CLI and test the function directly.

File: datacatalog/cloud-client/src/test/java/com/example/datacatalog/LookupEntryBigQueryDatasetTest.java [hunk 5]
```diff
@@ -0,0 +1,68 @@
+/*
```
Comment (kurtisvg) [inline datacatalog/cloud-client/src/test/java/com/example/datacatalog/LookupEntryBigQueryDatasetTest.java]: Tests don't need to be in separate files (unless you prefer them that way).

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java (hunk 1 above)
Comment (ricardosm-cit) [reply datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java]: Done.

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java (hunk 3 above)
Comment (ricardosm-cit) [reply datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java]: They are alternatives to do the same thing. I commented out the sqlResource part. Please let me know if it works.

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java (hunk 4 above)
Comment (ricardosm-cit) [reply datacatalog/cloud-client/src/main/java/com/example/datacatalog/LookupEntryBigQueryDataset.java]: Done.

File: datacatalog/cloud-client/src/test/java/com/example/datacatalog/LookupEntryBigQueryDatasetTest.java (hunk 5 above)
Comment (ricardosm-cit) [reply datacatalog/cloud-client/src/test/java/com/example/datacatalog/LookupEntryBigQueryDatasetTest.java]: Done.

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/QuickStart.java [hunk 6]
```diff
@@ -0,0 +1,4 @@
+package com.example.datacatalog;
+
+public class QuickStart {
```
Comment (kurtisvg) [inline datacatalog/cloud-client/src/main/java/com/example/datacatalog/QuickStart.java]: Is this file supposed to be here?

File: datacatalog/cloud-client/pom.xml [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+        </dependency>
+        <dependency>
+            <groupId>com.google.truth</groupId>
+            <artifactId>truth</artifactId>
+            <version>0.42</version>
+            <scope>test</scope>
+        </dependency>
+    </dependencies>
+
+    <build>
```
Comment (kurtisvg) [inline datacatalog/cloud-client/pom.xml]: I think this section is probably unnecessary with CLI removed.

File: datacatalog/cloud-client/src/main/java/com/example/datacatalog/QuickStart.java (hunk 6 above)
Comment (ricardosm-cit) [reply datacatalog/cloud-client/src/main/java/com/example/datacatalog/QuickStart.java]: Sorry, it's a WIP accidentally pushed. Just removed it...

File: datacatalog/cloud-client/pom.xml (hunk 7 above)
Comment (ricardosm-cit) [reply datacatalog/cloud-client/pom.xml]: Agree. Done.

**Comment by kurtisvg** [comment]
fixes #1410

**Pull request merged**